apollo-router/tests/snapshots/apollo_reports__stats.snap

```diff
@@ -34,7 +34,52 @@ traces_per_query:
           registered_operation_count: 0
           forbidden_operation_count: 0
           requests_without_field_instrumentation: 0
-        per_type_stat: {}
+        per_type_stat:
+          "":
+            per_field_stat:
+              "":
+                return_type: ""
```

Comment:
Empty strings here and below look unexpected. Or is it because `apollo-router/tests/apollo_reports.rs` creates incomplete artificial data?

Reply:
These do look unexpected. As this is an integration test I think this is a bug.

Reply:
The query from this test:

```graphql
query{topProducts{name reviews {author{name}} reviews{author{name}}}}
```

Lightly edited `dbg!` output for one of the the FTV1 trace:

<details>

```rust
root: Some(Node {
    original_field_name: "",
    r#type: "",
    parent_type: "",
    id: None,
    child: [
        Node {
            original_field_name: "",
            r#type: "[Product]",
            parent_type: "Query",
            id: Some(ResponseName("topProducts"))
            child: [
                Node {
                    original_field_name: "",
                    r#type: "",
                    parent_type: "",
                    id: Some(Index(0)),
                    child: [
                        Node {
                            original_field_name: "",
                            r#type: "String!",
                            parent_type: "Product",
                            child: [],
                            id: Some(ResponseName("upc")),
                        },
                        Node {
                            original_field_name: "",
                            r#type: "String",
                            parent_type: "Product",
                            child: [],
                            id: Some(ResponseName("name")),
                        },
                    ],
                },
                Node {
                    original_field_name: "",
                    r#type: "",
                    parent_type: "",
                    id: Some(Index(1)),
                    child: [
                        Node {
                            original_field_name: "",
                            r#type: "String!",
                            parent_type: "Product",
                            child: [],
                            id: Some(ResponseName("upc")),
                        },
                        Node {
                            original_field_name: "",
                            r#type: "String",
                            parent_type: "Product",
                            child: [],
                            id: Some(ResponseName("name")),
                        },
                    ],
                },
                Node {
                    original_field_name: "",
                    r#type: "",
                    parent_type: "",
                    id: Some(Index(2)),
                    child: [
                        Node {
                            original_field_name: "",
                            r#type: "String!",
                            parent_type: "Product",
                            child: [],
                            id: Some(ResponseName("upc")),
                        },
                        Node {
                            original_field_name: "",
                            r#type: "String",
                            parent_type: "Product",
                            child: [],
                            id: Some(ResponseName("name")),
                        },
                    ],
                },
            ],
```

</details>

My guesses:

* not all nodes should add an entry in a "per field" map, perhaps only those with `id = Some(ResponseName(something))`?
* `id` is the struct field to look at, not `original_field_name`. comments in the `.proto` file suggest that `original_field_name` is only used when a field is aliased

Reply:
My latest push does this, which make the data in test look much more plausible (no empty string key)